src/libutil/error.cc

```diff
@@ -18,7 +18,7 @@ void BaseError::addTrace(std::shared_ptr<Pos> && e, HintFmt hint)
 
 void throwExceptionSelfCheck(){
     // This is meant to be caught in initLibUtil()
-    throw SysError("C++ exception handling is broken. This would appear to be a problem with the way Nix was compiled and/or linked and/or loaded.");
+    throw Error("C++ exception handling is broken. This would appear to be a problem with the way Nix was compiled and/or linked and/or loaded.");
```

Comment:
We probably want to keep SysError here, right?

Reply:
Or `NativeSysError`?

Reply:
https://github.com/NixOS/nix/pull/10361